cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java

```diff
@@ -287,6 +287,23 @@ void sanityCheckConcurrency() {
     }
   }
 
+  /**
+   * Sanity check to ensure that
+   * <ul>
+   *   <li>{@link AnalyzerConfig#TOPIC_REPLICA_COUNT_BALANCE_MIN_GAP_CONFIG} <=
+   *   {@link AnalyzerConfig#TOPIC_REPLICA_COUNT_BALANCE_MIN_GAP_CONFIG}</li>
+   * </ul>
+   */
+  private void sanityCheckBalancingConstraints() {
+    int topicReplicaBalanceMinGap = getInt(AnalyzerConfig.TOPIC_REPLICA_COUNT_BALANCE_MIN_GAP_CONFIG);
+    int topicReplicaBalanceMaxGap = getInt(AnalyzerConfig.TOPIC_REPLICA_COUNT_BALANCE_MAX_GAP_CONFIG);
+    if (topicReplicaBalanceMinGap > topicReplicaBalanceMaxGap) {
```

Comment:
Should we allow `topicReplicaBalanceMinGap  == topicReplicaBalanceMaxGap`? In this case, the `computedLimit` is essentially ignored in the `gapBasedBalanceLimit` method since the return value of this method does not depend on `computedLimit`

Reply:
We intentionally allow equality of min/max gaps.

As per their discretion, this enables users to configure their goal such that each topic has deterministic limits on each broker that is independent of the percentage-based `computedLimit`.

Reply:
ok